### PR TITLE
feat(java-binding): support java binding on stream chunk

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

In this PR, we add java binding support to read rust `StreamChunk` from java. We can now access rust `StreamChunk` via `StreamChunkIterator.java`.

Besides, the PR involves the following changes.
- Introduce `StreamChunkRow` in rust, and combine `StreamChunkRow` and `KeyedRow` into an enum as the return type for all row types returned to java so that the `StreamChunkRow` and `KeyedRow` can share the same java binding method to access row. `BaseRow.java` is introduced to support the `get<Type>` method for `StreamChunkRow` to share row acess method with `KeyedRow`.
- Rename the previous `Iterator.java` to `HummockIterator.java` to make the name less general.
- Add demo ci for stream chunk iterator. To run the demo, run `cargo make run-java-binding-stream-chunk-demo`.
- Avoid using `mvn exec` to run the demos.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/java_binding/src/lib.rs [hunk 1]
```diff
@@ -217,7 +261,7 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_vnodeCount(
 }
 
 #[no_mangle]
-pub extern "system" fn Java_com_risingwave_java_binding_Binding_iteratorNew<'a>(
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_hummockIteratorNew<'a>(
```
Comment (Gun9niR) [inline src/java_binding/src/lib.rs:264]: Since the java side is renamed to `HummockIterator`, should the rust side get renamed also?

**Review by Gun9niR — APPROVED** [review]
LGTM. Just curious, what is the downside of `mvn exec`?

File: src/java_binding/src/lib.rs (hunk 1 above)
Comment (wenym1) [reply src/java_binding/src/lib.rs:264]: renamed to `HummockJavaBindingIterator`

**Comment by wenym1** [comment]
> Just curious, what is the downside of mvn exec?

Seems that mvn exec cannot redirect the stdin to the real java process.

**Review by yufansong — APPROVED** [review]

**Pull request merged**